Ducky-Chickens/bangazon-site
# Categories: Tickets #3 + #4

# Description
All content for `product_types` table is rendered to DOM. Additionally, `client` directory restructured and `express.static` implemented. 

## Related Ticket(s)
#3 + #4 

## Problem to Solve/Changes
+ Printing only 3 items per product type 
+ Scripting errors resolved with adjusted indentation and script blocks
+ `express.static` created in `app.js` to implement client environment and eliminate full stack traversing
+ All client-side Javascript files nested in `../client/js`

## Expected Behavior
+ When directed to `.../categories`, all product types listed with card formatting including, at most, 3 products per product type.
+ When clicking on product card, page redirects to `.../catergories/:id` and renders all products related to specified product type
+ When clicking product title, page redirects to `.../products/:id` and renders all specified product information.

## Steps to Test Solution

1. Run `node build-db.js`
1. Run `nodemon app.js`
1. Direct browser to `.../categories` to ensure all product types are listed
1. Click any/all product type(s); ensure browser is redirected to `.../catergories/:id` and appropriate products are rendered
1. Click any/all product title(s); ensure browser is redirected to `.../products/:id` and appropriate product information is rendered

## Documentation
